src/pages/transactions/TransactionForm.js

```diff
@@ -0,0 +1,158 @@
+import { useRef } from "react";
+import { useRouter } from "next/router";
+
+export default function TransactionForm({ onAddTransaction }) {
+  const router = useRouter();
+  const typeRef = useRef();
```

Comment:
I am not familiar with the 'useRef' hook. From what I have briefly read, it seems to be a sensible choice in this context,, but I haven't done in-depth research. In the future, when using functions, hooks or similar most of us are not familiar with, please comment briefly on why you use them and what they do.